kyo-core/shared/src/main/scala/kyo/System.scala

```diff
@@ -0,0 +1,142 @@
+package kyo
+
+import System.Parser
+import java.lang.System as JSystem
+import kyo.kernel.Reducible
+
+abstract class System:
+    def env[E, A](name: String)(using Parser[E, A], Frame): Maybe[A] < (Abort[E] & IO)
+    def property[E, A](name: String)(using Parser[E, A], Frame): Maybe[A] < (Abort[E] & IO)
+    def lineSeparator(using Frame): String < IO
+    def userName(using Frame): String < IO
+    def userHome(using Frame): String < IO
+
+end System
+
+object System:
+
+    val local = Local.init(live)
+
+    val live: System =
+        new System:
+            def env[E, A](name: String)(using p: Parser[E, A], frame: Frame): Maybe[A] < (Abort[E] & IO) =
+                IO {
+                    val value = JSystem.getenv(name)
+                    if value == null then Maybe.empty
+                    else p(value).map(Maybe(_))
+                }
+
+            def property[E, A](name: String)(using p: Parser[E, A], frame: Frame): Maybe[A] < (Abort[E] & IO) =
+                IO {
+                    val value = JSystem.getProperty(name)
+                    if value == null then Maybe.empty
+                    else p(value).map(Maybe(_))
+                }
+
+            def lineSeparator(using Frame): String < IO = IO(JSystem.lineSeparator())
+
+            def userName(using Frame): String < IO = IO(JSystem.getProperty("user.name"))
+
+            def userHome(using Frame): String < IO = IO {
+                Maybe(JSystem.getProperty("user.home"))
+                    .orElse(Maybe(JSystem.getenv("HOME")))
+                    .getOrElse {
+                        val osName = JSystem.getProperty("os.name").toLowerCase
+                        if osName.contains("win") then
+                            JSystem.getenv("USERPROFILE")
+                        else
+                            "/home/" + JSystem.getProperty("user.name")
+                        end if
+                    }
+            }
```

Comment:
We could provide this and perhaps more directories, based on https://github.com/dirs-dev/directories-jvm

Reply:
Good call! I've just realized that `Path` is a better place to put these methods. I'll remove them here and create a separate PR.